lib/Auth/Process/IsCesnetEligible.php

```diff
@@ -45,24 +52,39 @@ class IsCesnetEligible extends \SimpleSAML\Auth\ProcessingFilter
      */
     private $rpcConnector;
 
+    /**
+     * @var LdapConnector\
+     */
+    private $ldapConnector;
+
     public function __construct($config, $reserved)
     {
         parent::__construct($config, $reserved);
 
-        if (!isset($config[self::CESNET_ELIGIBLE_LAST_SEEN_ATTR])) {
+        if (!isset($config[self::RPC_ATTRIBUTE_NAME])) {
             throw new Exception(
                 "cesnet:IsCesnetEligible - missing mandatory configuration option '" .
-                self::CESNET_ELIGIBLE_LAST_SEEN_ATTR . "'."
+                self::RPC_ATTRIBUTE_NAME . "'."
             );
         }
 
         if (isset($config['attrName'])) {
-            $this->attrName = $config['attrName'];
+            $this->returnAttrName = $config['attrName'];
         } else {
-            $this->attrName = self::DEFAULT_ATTR_NAME;
+            $this->returnAttrName = self::DEFAULT_ATTR_NAME;
         }
 
-        $this->cesnetEligibleLastSeenAttr = $config[self::CESNET_ELIGIBLE_LAST_SEEN_ATTR];
+        $this->rpcAttrName = $config[self::RPC_ATTRIBUTE_NAME];
+        $this->interface = $config[self::INTERFACE_PROPNAME];
+
+        if ($this->interface === self::LDAP) {
+            $this->ldapAttrName = $config[self::LDAP_ATTRIBUTE_NAME];
+            if (!empty($this->ldapAttrName)) {
+                $this->ldapConnector = new AdapterLdap();
+            } else {
+                $this->interface = self::RPC;
```

Comment:
Please add warning that ldapAttrName is missing and RPC interface will be used

Reply:
Done